src/shapes/audio_token.class.js

```diff
@@ -0,0 +1,197 @@
+(function(global) {
+
+  'use strict';
+
+  if (!global.fabric) {
+    global.fabric = { };
+  }
+
+  if (global.fabric.Audio_token) {
+    fabric.warn('fabric.Audio_token is already defined.');
+    return;
+  }
+
+  /**
+   * Audio_token class
+   * @class fabric.Audio_token
+   * @extends fabric.Object
+   * @see {@link fabric.Audio_token#initialize} for constructor definition
+   */
+  fabric.Audio_token = fabric.util.createClass(fabric.Object, /** @lends fabric.Audio_token.prototype */ {
+
+    /**
+     * Type of an object
+     * @type String
+     * @default
+     */
+    type: 'audio_token',
+
+    /**
+     * unique key - used for now just to confirm play/pause for specific tokens
+     * @since 2.0.0
+     * @type String
+     * @default
+     */
+    cacheKey: '',
+
+    /**
+     * Indicates whether audio is in 'play' mode
+     * @type Boolean
+     * @default
+     */
+    isPlaying: false,
+
+    /**
+     * Indicates whether audio is in 'paused' mode
+     * @type Boolean
+     * @default
+     */
+    isPaused: false,
```

Comment:
Its not clear to me why we need both isPlaying and isPaused. Is seems we only have 2 states here... playing and not. Am I missing something?

And if we do have more... would an enum make more sense?

Reply:
We definitely want to distinguish between 'not playing' and 'paused' - to determine if we want to call audio.play() or audio.resume()   (I don't know how the audio calls will look but this is the way other audio libs I have used in the past have worked). I'll change this to an enum though to avoid ever having a situation where both are somehow true. (STOPPED, PLAYING, PAUSED, and eventually maybe PENDING_FILE_LOAD or something)

Reply:
Ah got it.. I didn't see it used but I guess it wouldn't be needed until the sound lib was being used.

For the enum vs 2 flags... really up to you, but it helps me personally better understand the various states.

Reply:
Yeah right now they are just used in the controls mixin - and just for debugging the states. I can come up with arguments on either side of enums vs a small set of bools. It would be a simple change so maybe I will wait until we start using them and see which feels better in practice.

Reply:
Oh... also used in the render to determine the img src of the play button :P